In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Closed_deals

In [4]:
# carga del dataset y muestreo de 10 registros aleatorios
Closed_deals = pd.read_csv('Datasets_original/olist_closed_deals_dataset.csv')
Closed_deals.sample(10)

mql_id                         seller_id  \
627  f2e5f6f0caaa54cb5161d5cd753cf57f  d622a61456dcbf4cc960221e15ac48b7   
312  bf5b131e6aee15fa373fb4d113b4a65c  8a8835a43bda99fd37310ad76c457510   
336  e84d3444422cda735543114cf5df6b95  db2956745b3a8e9f3785c99f34b5d25e   
11   408a9c4a79800232ac656249af3162eb  b566ab0ef88016e00422755e305103c6   
776  b466c22d12d52c8a0d3792d2eb44b828  623ad9d0c1e14bb5464a8c72bff7b4af   
789  b14ab177881b4c06b81317c5f2034ede  8fb67a334bacec338885b4e9e1708073   
373  99fc32e3c05e03597a692c4fd9a9d162  6a38087bc8ad4f89ff453561005f6dea   
645  5d23750fc8ccf6284ca2ef9063c7b395  1e26c56af8d48424e1df0140a2401f83   
240  f63f7107398c54c17b5d14eadfa76842  bc2579cde838eb0fa8477f7efb1298d1   
633  8e8336b19eec4f492acff4a97a5a4bf4  9140ef75cc8211c4035935e80e567204   

                               sdr_id                             sr_id  \
627  56bf83c4bb35763a51c2baab501b4c67  4ef15afb4b2723d8f3d81e51ec7afefe   
312  34d40cdaf94010a1d05b0d6212f9e909  fbf4aef3f6915dc0c3c97d6812522f6a   
336  9d12ef1a7eca3ec58c545c678af7869c  de63de0d10a6012430098db33c679b0b   
11   de63de0d10a6012430098db33c679b0b  d3d1e91a157ea7f90548eef82f1955e3   
776  4b339f9567d060bcea4f5136b9f5949e  6565aa9ce3178a5caf6171827af3a9ba   
789  9d12ef1a7eca3ec58c545c678af7869c  fbf4aef3f6915dc0c3c97d6812522f6a   
373  b90f87164b5f8c2cfa5c8572834dbe3f  4ef15afb4b2723d8f3d81e51ec7afefe   
645  9e4d1098a3b0f5da39b0bc48f9876645  fbf4aef3f6915dc0c3c97d6812522f6a   
240  34d40cdaf94010a1d05b0d6212f9e909  068066e24f0c643eb1d089c7dd20cd73   
633  34d40cdaf94010a1d05b0d6212f9e909  4ef15afb4b2723d8f3d81e51ec7afefe   

                won_date                 business_segment      lead_type  \
627  2018-02-28 18:22:11                             baby        offline   
312  2018-05-17 12:09:33                  car_accessories     online_big   
336  2018-03-22 19:35:04          audio_video_electronics  online_medium   
11   2018-02-23 13:54:40  construction_tools_house_garden     online_big   
776  2018-02-27 17:39:50              household_utilities     online_big   
789  2018-04-27 12:18:34                       home_decor  online_medium   
373  2018-05-11 21:26:07                        computers     online_big   
645  2018-02-26 13:02:58  construction_tools_house_garden        offline   
240  2018-05-10 20:10:08              household_utilities        offline   
633  2018-06-05 18:00:37                              pet       industry   

    lead_behaviour_profile has_company has_gtin average_stock business_type  \
627                  eagle         NaN      NaN           NaN      reseller   
312                    cat         NaN      NaN           NaN      reseller   
336                    cat         NaN      NaN           NaN  manufacturer   
11                     cat         NaN      NaN           NaN      reseller   
776                   wolf         NaN      NaN           NaN  manufacturer   
789                    cat         NaN      NaN           NaN  manufacturer   
373                    cat         NaN      NaN           NaN      reseller   
645                    cat         NaN      NaN           NaN      reseller   
240                    cat         NaN      NaN           NaN      reseller   
633                  eagle         NaN      NaN           NaN  manufacturer   

     declared_product_catalog_size  declared_monthly_revenue  
627                            NaN                       0.0  
312                            NaN                       0.0  
336                            NaN                       0.0  
11                             NaN                       0.0  
776                            NaN                       0.0  
789                            NaN                       0.0  
373                            NaN                       0.0  
645                            NaN                       0.0  
240                            NaN                       0.0  
633                            NaN        

In [6]:
# tamaño del dataset
Closed_deals.shape

(842, 14)

In [5]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Closed_deals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 842 entries, 0 to 841
Data columns (total 14 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   mql_id                         842 non-null    object 
 1   seller_id                      842 non-null    object 
 2   sdr_id                         842 non-null    object 
 3   sr_id                          842 non-null    object 
 4   won_date                       842 non-null    object 
 5   business_segment               841 non-null    object 
 6   lead_type                      836 non-null    object 
 7   lead_behaviour_profile         665 non-null    object 
 8   has_company                    63 non-null     object 
 9   has_gtin                       64 non-null     object 
 10  average_stock                  66 non-null     object 
 11  business_type                  832 non-null    object 
 12  declared_product_catalog_size  69 non-null     flo

In [6]:
# descripcion de las columnas numéricas
Closed_deals.describe()

declared_product_catalog_size  declared_monthly_revenue
count                      69.000000              8.420000e+02
mean                      233.028986              7.337768e+04
std                       352.380558              1.744799e+06
min                         1.000000              0.000000e+00
25%                        30.000000              0.000000e+00
50%                       100.000000              0.000000e+00
75%                       300.000000              0.000000e+00
max                      2000.000000              5.000000e+07

In [7]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariciones
Closed_deals.describe(include=['O'])

mql_id                         seller_id  \
count                                842                               842   
unique                               842                               842   
top     5420aad7fec3549a85876ba1c529bd84  2c43fb513632d29b3b58df74816f1b06   
freq                                   1                                 1   

                                  sdr_id                             sr_id  \
count                                842                               842   
unique                                32                                22   
top     4b339f9567d060bcea4f5136b9f5949e  4ef15afb4b2723d8f3d81e51ec7afefe   
freq                                 140                               133   

                   won_date business_segment      lead_type  \
count                   842              841            836   
unique                  824               33              8   
top     2018-05-04 03:00:00       home_decor  online_medium   
freq                      6              105            332   

       lead_behaviour_profile has_company has_gtin average_stock business_type  
count                     665          63       64            66           832  
unique                      9           2        2             6             3  
top                       cat        True     True          5-20      reseller  
freq                      407          58       54            22           587

In [18]:
# cuantos datos nulos por columna se encuentran
Closed_deals.isna().sum()*100/Closed_deals.shape[0]

mql_id                            0.000000
seller_id                         0.000000
sdr_id                            0.000000
sr_id                             0.000000
won_date                          0.000000
business_segment                  0.118765
lead_type                         0.712589
lead_behaviour_profile           21.021378
has_company                      92.517815
has_gtin                         92.399050
average_stock                    92.161520
business_type                     1.187648
declared_product_catalog_size    91.805226
declared_monthly_revenue          0.000000
dtype: float64

- De un total de 842 registros las columnas has_company, has_gtin, average_stock y declared_product_catalog_size tiene mas del 90 % de sus datos faltantes
- Esta tabla no aporta datos muy relevantes

## Customers

In [13]:
# carga del dataset y muestreo de 10 registros aleatorios
Customers = pd.read_csv('Datasets_original/olist_customers_dataset.csv')
Customers.sample(10)

customer_id                customer_unique_id  \
45711  75119e7aa32e3e42fbd019b09af7e7e1  b6ed057c95d8ed2ca7df910ef50674c1   
86192  8bad98b4cfc0bec348b202ea08123680  9764c899917199574678b4d3f1aacfa3   
33525  aa4e8db1608174fbd06df307c6ff687c  d7e0432688bbd72d3567568903f6117c   
30676  2c4c10219e88c96182dd31e470b89c6d  ebc2a17ea966619e7087a1ccaefc0b65   
47172  502bb2b647bf2cd2b55e73381cb9699c  88607ce3a266c48ee56db9695bff9dd6   
22075  e443a16cb8c9de00c7fe871c07c722da  62571cfc1e7bc956d34304b50e67167c   
16079  cdb2d7e08c5243701392d8b232a8a6b9  892bdc0d449b798eec8375762fb411b9   
83677  b104d98edcfb89e1cda5cf1510faecf1  28e42f3e120527d138764c5244f2bfb4   
48698  ba15d2abe9d748ca9cae18cd1cced72c  b69d80e77b9f6172a5fe86c1979dbc6e   
17902  39d5be86f1b2c886e37fde07eaffd537  1b9c0951f6e51a50af8587aac775139b   

       customer_zip_code_prefix  customer_city customer_state  
45711                     26520      nilopolis             RJ  
86192                      1501      sao paulo             SP  
33525                     12120       tremembe             SP  
30676                      5778      sao paulo             SP  
47172                      6436        barueri             SP  
22075                     60743      fortaleza             CE  
16079                     26582       mesquita             RJ  
83677                     90030   porto alegre             RS  
48698                     95765  bom principio             RS  
17902                     17210            jau             SP

In [14]:
# tamaño del dataset
Customers.shape

(99441, 5)

In [12]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Customers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


In [13]:
# descripcion de las columnas numéricas 
Customers.describe()

customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

In [14]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Customers.describe(include=['O'])

customer_id                customer_unique_id  \
count                              99441                             99441   
unique                             99441                             96096   
top     06b8999e2fba1a1fbc88172c00ba8bc7  8d50f5eadf50201ccdcedfb9e2ac8455   
freq                                   1                                17   

       customer_city customer_state  
count          99441          99441  
unique          4119             27  
top        sao paulo             SP  
freq           15540          41746

- Analizando Customers faltan datos de customers_unique_id que no hay referencia de como fueron creados esos codigos
- Esta tabla se puede utilizar para saber de qué ciudades provienen los consumos

## Geolocation

In [16]:
# carga del dataset y muestreo de 10 registros aleatorios
Geolocation = pd.read_csv('Datasets_original/olist_geolocation_dataset.csv')
Geolocation.sample(10)

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
556905                        30750       -19.899538       -43.976886   
715533                        53431        -7.903548       -34.836745   
156219                         6390       -23.540203       -46.847964   
400487                        19470       -21.766636       -52.112499   
529898                        29160       -20.230771       -40.274145   
332593                        14169       -21.141628       -47.977734   
537798                        29560       -20.770679       -41.673460   
977760                        96170       -31.362460       -51.971037   
232791                         9581       -23.637653       -46.581177   
690555                        45400       -13.371195       -39.075197   

           geolocation_city geolocation_state  
556905       belo horizonte                MG  
715533             paulista                PE  
156219          carapicuiba                SP  
400487  presidente epitacio                SP  
529898                serra                ES  
332593          sertãozinho                SP  
537798               guacui                ES  
977760  são lourenço do sul                RS  
232791   sao caetano do sul                SP  
690555              valenca                BA

In [18]:
# tamaño del dataset
Geolocation.shape

(1000163, 5)

In [19]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Geolocation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


In [20]:
# descripcion de las columnas numéricas
Geolocation.describe()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng
count                 1.000163e+06     1.000163e+06     1.000163e+06
mean                  3.657417e+04    -2.117615e+01    -4.639054e+01
std                   3.054934e+04     5.715866e+00     4.269748e+00
min                   1.001000e+03    -3.660537e+01    -1.014668e+02
25%                   1.107500e+04    -2.360355e+01    -4.857317e+01
50%                   2.653000e+04    -2.291938e+01    -4.663788e+01
75%                   6.350400e+04    -1.997962e+01    -4.376771e+01
max                   9.999000e+04     4.506593e+01     1.211054e+02

In [21]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Geolocation.describe(include=['O'])

geolocation_city geolocation_state
count           1000163           1000163
unique             8011                27
top           sao paulo                SP
freq             135800            404268

- Esta tabla puede servir para ubicar las ciudades correctamente en objeto mapa del Dashboard.

## Marketing

In [15]:
# carga del dataset y muestreo de 10 registros aleatorios
Marketing = pd.read_csv('Datasets_original/olist_marketing_qualified_leads_dataset.csv')
Marketing.sample(10)

mql_id first_contact_date  \
2394  9b46cae14639ee92bc8477140c8cf2f1         2018-03-06   
6686  02b0c69e363112f8c2d553a48a894db8         2018-03-05   
7469  1785cff273aedbd875c814b46928ad7f         2018-05-08   
7259  92dc4a66a118301843f185df549d40bf         2018-02-15   
7719  ae6272b0342a76d5a6ba771b94ac048c         2018-04-09   
5930  1a54ed0d8ca0bdb0cf54794977feb05a         2018-05-28   
4273  ff5a32529137a0ce614e6a37307423c5         2017-10-28   
3934  879076927f643b14ada250ec07ef4a88         2018-03-18   
877   afac78e614dac1a1470030ea55c08eef         2018-02-06   
6278  6179c371b2b9c0705516308bf05fb5c3         2018-04-25   

                       landing_page_id          origin  
2394  120ad10683075526ee1b3b00489dafec           email  
6686  22c29808c4f815213303f8933030604c  organic_search  
7469  22c29808c4f815213303f8933030604c             NaN  
7259  88740e65d5d6b056e0cda098e1ea6313          social  
7719  120ad10683075526ee1b3b00489dafec     paid_search  
5930  73fec6021437e723ff5adae8dff89e23  organic_search  
4273  22c29808c4f815213303f8933030604c           other  
3934  22c29808c4f815213303f8933030604c  organic_search  
877   22c29808c4f815213303f8933030604c     paid_search  
6278  b76ef37428e6799c421989521c0e5077         unknown

In [25]:
# tamaño del dataset
Marketing.shape

(8000, 4)

In [26]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Marketing.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8000 entries, 0 to 7999
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   mql_id              8000 non-null   object
 1   first_contact_date  8000 non-null   object
 2   landing_page_id     8000 non-null   object
 3   origin              7940 non-null   object
dtypes: object(4)
memory usage: 250.1+ KB


In [27]:
# descripcion de las columnas numéricas
Marketing.describe()

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [28]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Marketing.describe(include=['O'])

mql_id first_contact_date  \
count                               8000               8000   
unique                              8000                336   
top     dac32acd4db4c29c230538b72f8dd87d         2018-05-02   
freq                                   1                 93   

                         landing_page_id          origin  
count                               8000            7940  
unique                               495              10  
top     b76ef37428e6799c421989521c0e5077  organic_search  
freq                                 912            2296

In [17]:
# cuantos datos nulos por columna se encuentran
Marketing.isna().sum()*100/Marketing.shape[0]

mql_id                0.00
first_contact_date    0.00
landing_page_id       0.00
origin                0.75
dtype: float64

- Esta tabla no aporta datos significativos para el análisis que queremos realizar

## Order_items

In [85]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_items = pd.read_csv('Datasets_original/olist_order_items_dataset.csv')
Order_items.sample(10)

order_id  order_item_id  \
97638   dd845e1cdb19f08d029a727adc9a5a77              1   
104119  ec7c6bfed4adf361322a0b6d6e5bfc51              1   
102620  e90e7c747cb478cb9b973c929c0994e5              1   
106124  f12ff3c8b8e4945b4d0331a56df6ef85              1   
62771   8f50cc58ca77e14764fb138fcbde2b5c              1   
42712   614990b55e705d163e360ad21ed63a30              1   
54140   7b0c4136f285a86a8ceed54be6f56641              1   
80334   b6c73ab7a7655c4782eafca08b51ba55              1   
51790   75c493c31566f1a5d15fb9d248d5e4a4              1   
96004   d9d4a02bb9f1087deec1dae728740cdb              1   

                              product_id                         seller_id  \
97638   6d2928252aa5ba6a5fc439571dd0d748  ce27a3cc3c8cc1ea79d11e561e9bebb6   
104119  5129530d6042dd13060c2fd1748758d7  8a2d12a90e92f012ce5226fc6592b653   
102620  06649da52cbfbb4e9b6cfc332d56c119  ce248b21cb2adc36282ede306b7660e5   
106124  0a8699af8de9f3c25f80fe9b530df153  79ebd9a61bac3eaf882805ed4ecfa12a   
62771   6f33a4a09ae1180a0ee1ff4682b2d21f  725c32fa80c2faacc4fc88450d27314e   
42712   08a60a2b8d7362c5da1b6bc61f7d1191  6d66611d7c44cc30ce351abc49a68421   
54140   ee3d532c8a438679776d222e997606b3  f457c46070d02cadd8a68551231220dd   
80334   6bb50785f82871dde37ed1666da54bf3  4c2b230173bb36f9b240f2b8ac11786e   
51790   0f72996a8379314700a3ff9e822451e0  7ecd59e5e20407131822c1a68ac59c1f   
96004   83b00325c13c44245b2c3a2befa62a0e  ad781527c93d00d89a11eecd9dcad7c1   

        shipping_limit_date   price  freight_value  
97638   2016-10-14 12:02:00   58.00          17.33  
104119  2018-03-05 15:31:09  119.00          19.07  
102620  2018-06-14 15:54:21   79.90          18.44  
106124  2018-04-13 19:29:44  179.90          31.91  
62771   2017-10-20 10:28:19  139.99          17.55  
42712   2018-05-23 18:31:42  113.90          19.77  
54140   2018-02-08 02:53:59  100.00          12.83  
80334   2018-01-04 02:37:53   39.90          15.94  
51790   2018-02-15 10:08:53   49.99          16.92  
96004   2017-10-18 16:49:25  159.99          57.79

In [32]:
# tamaño del dataset
Order_items.shape

(112650, 7)

In [33]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_items.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


In [44]:
# descripcion de las columnas numéricas
Order_items.describe()

order_item_id          price  freight_value    order_total
count  112650.000000  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320     136.686539
std         0.705124     183.633928      15.806405     217.468683
min         1.000000       0.850000       0.000000       0.850000
25%         1.000000      39.900000      13.080000      45.000000
50%         1.000000      74.990000      16.260000      84.495000
75%         1.000000     134.900000      21.150000     149.900000
max        21.000000    6735.000000     409.680000   13440.000000

In [35]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_items.describe(include=['O'])

order_id                        product_id  \
count                             112650                            112650   
unique                             98666                             32951   
top     8272b63d03f5f79c56e9e4120aec44ef  aca2eb7d00ea1a7b8ebd4e68314663af   
freq                                  21                               527   

                               seller_id  shipping_limit_date  
count                             112650               112650  
unique                              3095                93318  
top     6560211a19b47992c3666cc44a7e94c0  2017-07-21 18:25:23  
freq                                2033                   21

In [70]:
# Revisión de la columna order_item_id; expresa la cantidad incremental de artículos presentes en una misma orden
Order_items[['order_id', 'order_item_id']].groupby(['order_id']).count().sort_values('order_item_id', ascending=False)

order_item_id
order_id                                       
8272b63d03f5f79c56e9e4120aec44ef             21
1b15974a0141d54e36626dca3fdc731a             20
ab14fdcfbe524636d65ee38360e22ce8             20
9ef13efd6949e4573a18964dd1bbe7f5             15
428a2f660dc84138d969ccd69a0ab6d5             15
...                                         ...
5a0911d70c1f85d3bed0df1bf693a6dd              1
5a082b558a3798d3e36d93bfa8ca1eae              1
5a07264682e0b8fbb3f166edbbffc6e8              1
5a071192a28951b76774e5a760c8c9b7              1
fffe41c64501cc87c801fd61db3f6244              1

[98666 rows x 1 columns]

In [84]:
# Revisión de la columna product_id; cuál es el producto más vendido
Order_items[['order_id','product_id']].groupby(['product_id']).count().sort_values('order_id', ascending=False)

order_id
product_id                                
aca2eb7d00ea1a7b8ebd4e68314663af       527
99a4788cb24856965c36a24e339b6058       488
422879e10f46682990de24d770e7f83d       484
389d119b48cf3043d311335e499d9c6b       392
368c6c730842d78016ad823897a372db       388
...                                    ...
6ca99f05e9c1952a6597264e87f01378         1
6ca54e191592a843087a042b0665d7ef         1
6ca5428af5edb5066b1966ad58501b60         1
6ca32b4631431cb1fe976880797a0f5c         1
fffe9eeff12fcbd74a2f2b007dde0c58         1

[32951 rows x 1 columns]

In [75]:
# Revisión de la columna product_id; cuánto facturó el producto más vendido
Order_items[['price','product_id']].groupby(['product_id']).sum().sort_values('price', ascending=False)

price
product_id                                
bb50f2e236e5eea0100680137654686c  63885.00
6cdd53843498f92890544667809f1595  54730.20
d6160fb7873f184099d9bc95e30376af  48899.34
d1c427060a0f73f6b889a5c7c61f2ac4  47214.51
99a4788cb24856965c36a24e339b6058  43025.56
...                                    ...
2e8316b31db34314f393806fd7b6e185      2.99
680cc8535be7cc69544238c1d6a83fe8      2.90
8a3254bee785a526d548a81a9bc3c9be      2.55
310dc32058903b6416c71faff132df9e      2.29
46fce52cef5caa7cc225a5531c946c8b      2.20

[32951 rows x 1 columns]

In [74]:
# Revisión de la columna seller_id; cuál es el vendedor con más órdenes
Order_items[['order_id','seller_id']].groupby(['seller_id']).count().sort_values('order_id', ascending=False)

order_id
seller_id                                 
6560211a19b47992c3666cc44a7e94c0      2033
4a3ca9315b744ce9f8e9374361493884      1987
1f50f920176fa81dab994f9023523100      1931
cc419e0650a3c5ba77189a1882b7556a      1775
da8622b14eb17ae2831f4ac5b9dab84a      1551
...                                    ...
5f57db27027655e6c6a391601daa0258         1
5e25c632f9db6cfb8bc100421c2737b6         1
5e106d93b717c2682b9383da480364a1         1
d36b07fd3c5088e477a72d1fccfe29d4         1
7d81e74a4755b552267cd5e081563028         1

[3095 rows x 1 columns]

In [72]:
# Revisión de la columna seller_id; cuánto sumaron las órdenes de cada vendedor
Order_items[['price','seller_id']].groupby(['seller_id']).sum().sort_values('price', ascending=False)

price
seller_id                                  
4869f7a5dfa277a7dca6462dcf3b52b2  229472.63
53243585a1d6dc2643021fd1853d8905  222776.05
4a3ca9315b744ce9f8e9374361493884  200472.92
fa1c13f2614d7b5c4749cbc52fecda94  194042.03
7c67e1448b00f6e969d365cea6b010ab  187923.89
...                                     ...
34aefe746cd81b7f3b23253ea28bef39       8.00
702835e4b785b67a084280efca355756       7.60
1fa2d3def6adfa70e58c276bb64fe5bb       6.90
77128dec4bec4878c37ab7d6169d6f26       6.50
cf6f6bc4df3999b9c6440f124fb2f687       3.50

[3095 rows x 1 columns]

## Order_payments

In [ ]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_payments = pd.read_csv('Datasets_original/olist_order_payments_dataset.csv');

In [31]:
Order_payments.sample(10)

order_id  payment_sequential payment_type  \
81374  e328e23027b916218c9fba2f6cf9b682                   1       boleto   
75082  d2b401a47dcec4ec39bc080b7154cd39                   1       boleto   
50237  9908ae99065b70b8df618def52474f12                   1  credit_card   
34363  c5c82d0544e75ee9ca884b6f458dfaba                   1  credit_card   
68617  f816e2d1a8c3f47837c88fc08ba39dce                   1       boleto   
19131  0e069a67e95068b8613f728bac35f6cc                   1  credit_card   
52065  84ce3bfa332a0a5e57045f91f6dce21e                   1  credit_card   
10883  add3e322151e47705826b60fbe20356f                   1  credit_card   
74328  66c1172ab0ddb605d4012ce1d4339ffb                   1  credit_card   
85700  ccc0965b6e9996d22fca63fe790a3c30                   1  credit_card   

       payment_installments  payment_value  
81374                     1          54.62  
75082                     1         153.10  
50237                     3         177.49  
34363                     1          57.68  
68617                     1        1337.19  
19131                     1          56.78  
52065                     6         127.40  
10883                     2          26.00  
74328                     7         193.84  
85700                     1         162.19

In [39]:
# tamaño del dataset
Order_payments.shape

(103886, 5)

In [40]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_payments.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


In [41]:
# descripcion de las columnas numéricas
Order_payments.describe()

payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

In [42]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_payments.describe(include=['O'])

order_id payment_type
count                             103886       103886
unique                             99440            5
top     fa65dad1b0e818e3ccc5cb0e39231352  credit_card
freq                                  29        76795

- El comportamiento de los pagos. La utilidad dependerá de la propuesta del PO

## Order_reviews

In [21]:
# carga del dataset y muestreo de 10 registros aleatorios
Order_reviews = pd.read_csv('Datasets_original/olist_order_reviews_dataset.csv')
Order_reviews.sample(10)

review_id                          order_id  \
60912  ba5e3ce78056a895ba926a9d86b05430  500c994fe4cc79f9ec4d6ad8cf94a49f   
97091  c9800e80911eab8d790c6ed3d501fcbb  e0429c361ab5500f38ecdc7a37f99955   
34854  14577c77a3c5dfcd138fc3d90d46507d  a03080d29751000dd3e2eab48835af82   
91091  4dd9adff34984d053150c0bb19cf74b1  379fed4a1ab7b5a3e8126fe0b76b9ebe   
91327  0fc4a0d2a0986e0a0fd093de7f495721  71038e0d9a6cdb865f447e4609420028   
98270  74b7783de7bcad29be5d3f067e021a79  cba5e2a02a8613805d4478be468a9a7a   
83814  9772bda6ba0443991e5f21939e8fb95a  b55b34e107238df26137bc117f892827   
41232  a7d268faaba30fd06c71947f6522ff2f  977864c01ce46bb65d6e13a345804645   
21368  59e15e5561115b91572e33e90dd77e78  fcc8173153ab317f1827da42e62661f6   
78235  df7a6abd75766f9250b6ab55c8c9b24d  d21a08b18679aca2a35ed7fe46c513d1   

       review_score review_comment_title  \
60912             4                  NaN   
97091             3                  NaN   
34854             4                  NaN   
91091             3                  NaN   
91327             5                  NaN   
98270             1                  NaN   
83814             5                  NaN   
41232             5                  NaN   
21368             5                  NaN   
78235             1                  NaN   

                                  review_comment_message review_creation_date  \
60912                                                NaN  2018-06-08 00:00:00   
97091                                                NaN  2018-04-21 00:00:00   
34854                                                NaN  2018-04-27 00:00:00   
91091                                                NaN  2018-08-01 00:00:00   
91327                                                NaN  2018-04-29 00:00:00   
98270                                                NaN  2018-02-28 00:00:00   
83814                                                NaN  2017-12-12 00:00:00   
41232                                                NaN  2018-07-26 00:00:00   
21368  O produto é excelente.São as melhores velas do...  2018-02-03 00:00:00   
78235  Não consigo falar com ninguém nas lojas lannis...  2017-06-24 00:00:00   

      review_answer_timestamp  
60912     2018-06-09 14:55:18  
97091     2018-04-22 00:35:48  
34854     2018-04-30 16:51:22  
91091     2018-08-04 12:29:42  
91327     2018-04-30 22:56:05  
98270     2018-02-28 16:29:43  
83814     2017-12-12 21:11:57  
41232     2018-07-31 19:09:43  
21368     2018-02-03 20:45:00  
78235     2017-06-24 17:19:09

In [46]:
# tamaño del dataset
Order_reviews.shape

(99224, 7)

In [47]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Order_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   review_id                99224 non-null  object
 1   order_id                 99224 non-null  object
 2   review_score             99224 non-null  int64 
 3   review_comment_title     11568 non-null  object
 4   review_comment_message   40977 non-null  object
 5   review_creation_date     99224 non-null  object
 6   review_answer_timestamp  99224 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


In [48]:
# descripcion de las columnas numéricas
Order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [49]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Order_reviews.describe(include=['O'])

review_id                          order_id  \
count                              99224                             99224   
unique                             98410                             98673   
top     7b606b0d57b078384f0b58eac1d41d78  c88b1d1b157a9999ce368f218a407141   
freq                                   3                                 3   

       review_comment_title review_comment_message review_creation_date  \
count                 11568                  40977                99224   
unique                 4527                  36159                  636   
top               Recomendo              Muito bom  2017-12-19 00:00:00   
freq                    423                    230                  463   

       review_answer_timestamp  
count                    99224  
unique                   98248  
top        2017-06-15 23:21:05  
freq                         4

In [22]:
# cuantos datos nulos por columna se encuentran
Order_reviews.isna().sum()*100/Order_reviews.shape[0]

review_id                   0.000000
order_id                    0.000000
review_score                0.000000
review_comment_title       88.341530
review_comment_message     58.702532
review_creation_date        0.000000
review_answer_timestamp     0.000000
dtype: float64

- La utilidad de esta tabla dependerá del objetivo propuesto.

## Orders

In [23]:
# carga del dataset y muestreo de 10 registros aleatorios
Orders = pd.read_csv('Datasets_original/olist_orders_dataset.csv')
Orders.sample(10)

order_id                       customer_id  \
44154  513f43ba9ea592cb81880aeeb11f08d4  9d951c0b8d3b39f81a4e35fd4aa21bf3   
12729  3ffa506c19577dce30f9dbce07028c38  42990531f1c563abd40f0a87b783c0c1   
11225  8d2a68ca0ac61329c56e0b74695b04b4  0ce2e9b14cff24808d5d4aa835c7252e   
30094  db0c4d3804fd0676152a2ac39cf5cc27  45f416cd82ad9c52c6ed085aabf3f203   
96331  16f391bc8fc37407de41720cd92f6266  ac5baec50a0a561a000ed259814803cb   
69790  70e218295bdf5efe77ddb143b1b3c0b4  409021a104c22b3c6f9fad25a36f26f5   
63313  03aba4d94f2bd838c0bb55be9d13b0f6  2daa54fc002886fa8a5deb96d1be1540   
2378   1ef47fb13ee9044b5e5aa3bd388755de  c1f326c1584e139615e3ba4ae4d23760   
95008  dc42fc68820b778654a001cb6eab8c96  42bdd174ffe332aeaf52cf9b556f8591   
87536  e2f3a90e59ee340336eda6e45ce3fb24  8fcb2ac952f183f3695080630fa8020c   

      order_status order_purchase_timestamp    order_approved_at  \
44154    delivered      2017-04-21 17:12:37  2017-04-25 08:22:55   
12729    delivered      2018-08-15 21:09:19  2018-08-15 21:23:53   
11225    delivered      2018-02-07 18:49:55  2018-02-07 19:10:13   
30094    delivered      2018-07-13 16:11:46  2018-07-13 16:31:26   
96331    delivered      2017-04-08 11:57:18  2017-04-08 13:05:15   
69790    delivered      2017-12-04 11:49:02  2017-12-04 16:32:07   
63313    delivered      2017-12-24 20:14:44  2017-12-24 20:26:42   
2378     delivered      2018-03-15 10:58:23  2018-03-15 11:10:29   
95008    delivered      2017-08-03 00:21:58  2017-08-03 00:30:09   
87536    delivered      2018-01-09 19:54:35  2018-01-09 20:11:36   

      order_delivered_carrier_date order_delivered_customer_date  \
44154          2017-04-25 12:35:36           2017-05-03 09:08:57   
12729          2018-08-17 14:36:00           2018-08-22 21:32:02   
11225          2018-02-09 22:18:46           2018-02-14 21:29:12   
30094          2018-07-16 14:10:00           2018-07-24 15:52:03   
96331          2017-04-25 11:47:32           2017-05-03 12:11:15   
69790          2017-12-06 19:08:42           2017-12-07 17:32:29   
63313          2017-12-28 19:14:17           2017-12-29 14:14:34   
2378           2018-03-15 23:58:52           2018-04-12 21:05:53   
95008          2017-08-03 18:46:50           2017-08-15 19:57:51   
87536          2018-01-10 20:52:49           2018-02-03 00:46:07   

      order_estimated_delivery_date  
44154           2017-05-11 00:00:00  
12729           2018-08-31 00:00:00  
11225           2018-03-07 00:00:00  
30094           2018-08-02 00:00:00  
96331           2017-05-04 00:00:00  
69790           2017-12-21 00:00:00  
63313           2018-01-15 00:00:00  
2378            2018-04-06 00:00:00  
95008           2017-08-29 00:00:00  
87536           2018-02-05 00:00:00

In [53]:
# tamaño del dataset
Orders.shape

(99441, 8)

In [54]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Orders.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


In [55]:
# descripcion de las columnas numéricas
Orders.describe()

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [56]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Orders.describe(include=['O'])

order_id                       customer_id  \
count                              99441                             99441   
unique                             99441                             99441   
top     e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
freq                                   1                                 1   

       order_status order_purchase_timestamp    order_approved_at  \
count         99441                    99441                99281   
unique            8                    98875                90733   
top       delivered      2018-04-11 10:48:14  2018-02-27 04:31:10   
freq          96478                        3                    9   

       order_delivered_carrier_date order_delivered_customer_date  \
count                         97658                         96476   
unique                        81018                         95664   
top             2018-05-09 15:48:00           2018-05-08 23:38:46   
freq                             47                             3   

       order_estimated_delivery_date  
count                          99441  
unique                           459  
top              2017-12-20 00:00:00  
freq                             522

In [24]:
# cuantos datos nulos por columna se encuentran
Orders.isna().sum()*100/Orders.shape[0]

order_id                         0.000000
customer_id                      0.000000
order_status                     0.000000
order_purchase_timestamp         0.000000
order_approved_at                0.160899
order_delivered_carrier_date     1.793023
order_delivered_customer_date    2.981668
order_estimated_delivery_date    0.000000
dtype: float64

- La utilidad de esta tabla dependerá del objetivo.

## Products

In [25]:
# carga del dataset y muestreo de 10 registros aleatorios
Products = pd.read_csv('Datasets_original/olist_products_dataset.csv')
Products.sample(10)

product_id        product_category_name  \
7086   ca230e32eccabf9d497e3776f6801f5f              cama_mesa_banho   
25353  48633254f8b7b802b1ca8c4f4dcf3e33                        artes   
17635  bb3542bf40cc975720ee44d03fcf0cd8           eletrodomesticos_2   
13633  95b7ae3076a4c91f203a0c3dd9a54b57                esporte_lazer   
16744  49af4c64c7a13a102a10c4c2d6cd4d91                     pet_shop   
21307  4a741a6cd8c7a4358ebfe566138fee78             moveis_decoracao   
7757   385b47aa4a6fda44c85e4899cfb81b96  fashion_bolsas_e_acessorios   
29849  deafa3faa1b80a845f4fbfb3ffa5d0b8                    papelaria   
15116  3846547667e1aa278c1d0259febee234                   automotivo   
18567  a4d2e285495aab9c4dfa00201bedcb87                    telefonia   

       product_name_lenght  product_description_lenght  product_photos_qty  \
7086                  51.0                       186.0                 1.0   
25353                 53.0                       638.0                10.0   
17635                 40.0                      3038.0                 1.0   
13633                 49.0                      1244.0                 1.0   
16744                 48.0                      1558.0                 2.0   
21307                 44.0                       999.0                 5.0   
7757                  33.0                       187.0                 5.0   
29849                 58.0                       503.0                 1.0   
15116                 60.0                       903.0                 5.0   
18567                 60.0                       562.0                 3.0   

       product_weight_g  product_length_cm  product_height_cm  \
7086             3250.0               29.0                7.0   
25353             200.0               25.0                5.0   
17635            2300.0               22.0               17.0   
13633             350.0               18.0               12.0   
16744             550.0               20.0               18.0   
21307           10550.0              104.0               14.0   
7757              100.0               16.0                5.0   
29849            5900.0               35.0               35.0   
15116            2200.0               30.0               25.0   
18567              50.0               16.0                3.0   

       product_width_cm  
7086               36.0  
25353              33.0  
17635              24.0  
13633              15.0  
16744              20.0  
21307              42.0  
7757               11.0  
29849              26.0  
15116              30.0  
18567              11.0

In [60]:
# tamaño del dataset
Products.shape

(32951, 9)

In [61]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Products.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


In [62]:
# descripcion de las columnas numéricas
Products.describe()

product_name_lenght  product_description_lenght  product_photos_qty  \
count         32341.000000                32341.000000        32341.000000   
mean             48.476949                  771.495285            2.188986   
std              10.245741                  635.115225            1.736766   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32949.000000       32949.000000       32949.000000   
mean        2276.472488          30.815078          16.937661   
std         4282.038731          16.914458          13.637554   
min            0.000000           7.000000           2.000000   
25%          300.000000          18.000000           8.000000   
50%          700.000000          25.000000          13.000000   
75%         1900.000000          38.000000          21.000000   
max        40425.000000         105.000000         105.000000   

       product_width_cm  
count      32949.000000  
mean          23.196728  
std           12.079047  
min            6.000000  
25%           15.000000  
50%           20.000000  
75%           30.000000  
max          118.000000

In [63]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Products.describe(include=['O'])

product_id product_category_name
count                              32951                 32341
unique                             32951                    73
top     1e9e8ef04dbcff4541ed26657ea517e5       cama_mesa_banho
freq                                   1                  3029

In [26]:
# cuantos datos nulos por columna se encuentran
Products.isna().sum()*100/Products.shape[0]

product_id                    0.000000
product_category_name         1.851234
product_name_lenght           1.851234
product_description_lenght    1.851234
product_photos_qty            1.851234
product_weight_g              0.006070
product_length_cm             0.006070
product_height_cm             0.006070
product_width_cm              0.006070
dtype: float64

- La utilidad de esta tabla dependerá del objetivo.

## Sellers

In [27]:
# carga del dataset y muestreo de 10 registros aleatorios
Sellers = pd.read_csv('Datasets_original/olist_sellers_dataset.csv')
Sellers.sample(10)

seller_id  seller_zip_code_prefix  \
425   2d42ebcda99fd55f6f4bc2f1ca035d50                   76400   
526   fac35d150712afb921ec59ac5f39bb11                   38650   
137   34aefe746cd81b7f3b23253ea28bef39                   81210   
1650  458d868c9642f55f18f04e951a3fdde6                   90670   
1912  cc63f0dd2acba93ffed4fe9f8e0321fa                   15025   
228   9d69ba0a7a626a64dc5db08e529559cf                   14075   
2562  412a4720f3e9431b4afa1476a1acddbe                   13060   
1932  9315e491484b1b0081ca166fc164108d                   86039   
2239  5d043cd5512d4bd2f88e5ccdd5736c38                    2710   
99    276677b5d08786d5dce7c2149dcce48b                   31730   

                seller_city seller_state  
425                  uruacu           GO  
526   bonfinopolis de minas           MG  
137                curitiba           PR  
1650           porto alegre           RS  
1912  sao jose do rio preto           SP  
228          ribeirao preto           SP  
2562               campinas           SP  
1932               londrina           PR  
2239              sao paulo           SP  
99           belo horizonte           MG

In [67]:
# tamaño del dataset
Sellers.shape

(3095, 4)

In [83]:
# revisamos la distribución de las ciudades (mejor con un gráfico)
Sellers.seller_city.value_counts()

sao paulo              694
curitiba               127
rio de janeiro          96
belo horizonte          68
ribeirao preto          52
                      ... 
taruma                   1
s jose do rio preto      1
domingos martins         1
messias targino          1
leme                     1
Name: seller_city, Length: 611, dtype: int64

In [68]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Sellers.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


In [69]:
# descripcion de las columnas numéricas
Sellers.describe()

seller_zip_code_prefix
count             3095.000000
mean             32291.059451
std              32713.453830
min               1001.000000
25%               7093.500000
50%              14940.000000
75%              64552.500000
max              99730.000000

In [70]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Sellers.describe(include=['O'])

seller_id seller_city seller_state
count                               3095        3095         3095
unique                              3095         611           23
top     3442f8959a84dea7ee197c632cb2df15   sao paulo           SP
freq                                   1         694         1849

In [80]:
# Revisamos uniendo las tablas, cuál es la ciudad con mayor cantidad de órdenes
Sellers.merge(Order_items)[['seller_city','order_id']].groupby(['seller_city']).count().sort_values('order_id', ascending=False)

order_id
seller_city               
sao paulo            27983
ibitinga              7750
curitiba              3016
santo andre           2964
belo horizonte        2593
...                    ...
sapiranga                1
janauba                  1
jarinu                   1
juzeiro do norte         1
04482255                 1

[611 rows x 1 columns]

In [82]:
# Revisamos uniendo las tablas, cuál es la ciudad con el monto mayor en ventas
Sellers.merge(Order_items)[['seller_city','price']].groupby(['seller_city']).sum().sort_values('price', ascending=False)

price
seller_city               
sao paulo       2702878.14
ibitinga         624592.94
curitiba         470759.82
rio de janeiro   358413.59
guarulhos        329494.38
...                    ...
resende              19.99
floranopolis         19.98
jarinu               14.63
macatuba             13.00
palotina              9.99

[611 rows x 1 columns]

## Product_category_name_translation

In [28]:
# carga del dataset y muestreo de 10 registros aleatorios
Product_category_name_translation = pd.read_csv('Datasets_original/product_category_name_translation.csv')
Product_category_name_translation.sample(10)

product_category_name          product_category_name_english
54                    artigos_de_natal                     christmas_supplies
36  construcao_ferramentas_ferramentas                costruction_tools_tools
48              livros_interesse_geral                 books_general_interest
22                          cool_stuff                             cool_stuff
7                utilidades_domesticas                             housewares
3                      cama_mesa_banho                         bed_bath_table
68                     fraldas_higiene                    diapers_and_hygiene
63         portateis_casa_forno_e_cafe  small_appliances_home_oven_and_coffee
50      fashion_underwear_e_moda_praia                fashion_underwear_beach
59                           cine_foto                             cine_photo

In [74]:
# tamaño del dataset
Product_category_name_translation.shape

(71, 2)

In [75]:
# tipo de datos, no nulos y cantidad de registros para cada columna
Product_category_name_translation.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


In [77]:
# como se comportan las variables categóricas-- count=cantidad, unique=registros unicos, top=mas veces registrado, freq=frecuencia de los registros con mas aparariociones
Product_category_name_translation.describe(include=['O'])

product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

# Carga en MySQL

In [79]:
from sqlalchemy.engine import create_engine
from urllib.parse import quote_plus as urlquote
from Pass_SQL import contraseña

In [80]:
str_conexion = 'mysql+pymysql://root:%s@localhost:3306/Proyecto_Grupal' % urlquote(contraseña())
conexion = create_engine(str_conexion)

In [ ]:
# Cargamos los dataset en una tabla
Product_category_name_translation.to_sql(name='product_category_name_translation', con=conexion)
Sellers.to_sql(name='sellers', con=conexion)
Products.to_sql(name='products', con=conexion)
Orders.to_sql(name='orders', con=conexion)
Order_reviews.to_sql(name='order_reviews', con=conexion)
Order_payments.to_sql(name='order_payments', con=conexion)
Order_items.to_sql(name='order_items', con=conexion)
Marketing.to_sql(name='marketing', con=conexion)
Geolocation.to_sql(name='geolocation', con=conexion)
Customers.to_sql(name='customers', con=conexion)
Closed_deals.to_sql(name='closed_deals', con=conexion);